core/src/main/scala/cats/mtl/Handle.scala

```diff
@@ -217,6 +218,41 @@ private[mtl] trait HandleInstances extends HandleLowPriorityInstances {
     }
 }
 
-object Handle extends HandleInstances {
+object Handle extends HandleInstances with HandleVariant {
+
   def apply[F[_], E](implicit ev: Handle[F, E]): Handle[F, E] = ev
+
+  def allowF[F[_], E]: AdHocSyntaxTired[F, E] =
+    new AdHocSyntaxTired[F, E]
+
+  final class AdHocSyntaxTired[F[_], E] {
```

Comment:
We can do tricks to avoid the allocations.
```suggestion
  final class AdHocSyntaxTired[F[_], E](private val unit: Unit) extends AnyVal {
```